addons/website_sale/models/website.py

```diff
@@ -550,7 +550,10 @@ def _send_abandoned_cart_email(self):
             (all_abandoned_carts - abandoned_carts).cart_recovery_email_sent = True
             for sale_order in abandoned_carts:
                 template = self.env.ref('website_sale.mail_template_sale_cart_recovery')
-                template.send_mail(sale_order.id, email_values=dict(email_to=sale_order.partner_id.email))
+                email_vals = {
+                    'email_to': sale_order.partner_id.email if not (template.email_to or template.partner_to) else False
```

Comment:
We generally try to keep lines below 100 chars in length in the sale scope, so we could do something like
```suggestion
                email_vals = {} if template.email_to or template.partner_to else {
                    'email_to': sale_order.partner_id.email
```